virtualization_arm64.h

```diff
@@ -65,7 +65,7 @@ void *newVZMacMachineIdentifierWithPath(const char *machineIdentifierPath);
 void *newVZMacMachineIdentifierWithBytes(void *machineIdentifierBytes, int len);
 nbyteslice getVZMacMachineIdentifierDataRepresentation(void *machineIdentifierPtr);
 
-VZMacOSRestoreImageStruct convertVZMacOSRestoreImage2Struct(VZMacOSRestoreImage *restoreImage);
+VZMacOSRestoreImageStruct convertVZMacOSRestoreImage2Struct(void *restoreImagePtr);
```

Comment:
This is also related to https://github.com/Code-Hex/vz/pull/48#discussion_r982274630

Reply:
This code is used only internally. So I don't think this is no problem

Reply:
Please revert this~

Reply:
I still think we need this ( https://github.com/Code-Hex/vz/pull/48#issuecomment-1260760961 ). It is really useful for the maintainance of macOS 11/macOS 10 compat code. If there are too many warnings, we will not notice the new ones.

It also makes the code consistent with the other `convert*` functions:
```go
virtualization.h:VZVirtioSocketConnectionFlat convertVZVirtioSocketConnection2Flat(void *connection);
virtualization_arm64.h:VZMacOSConfigurationRequirementsStruct convertVZMacOSConfigurationRequirements2Struct(void *requirementsPtr);
virtualization_arm64.h:VZMacHardwareModelStruct convertVZMacHardwareModel2Struct(void *hardwareModelPtr);
```
They all use void*

Reply:
@cfergeau Ping
(Because I don't need to support os 10)

Reply:
This specific change is about macOS11. Commit log mentions `CGO_CFLAGS=-mmacosx-version-min=11.0` and `warning: 'VZMacOSRestoreImage' is only available on macOS 12.0 or newer`.

I've reverted the vsock one (to avoid force push). I'd like to squash all the latest changes/remove the problematic commit and then force-push though as it's getting messy ^^.

Reply:
> This function is used only internally not in the Go world.

I know it is only internal. But the objc compiler warns about it.

> (because this is not following other convert function rules).

All other convert functions are using `void *`:
- https://github.com/Code-Hex/vz/blob/37646e120f85b7fd99989e7e4b23cf068564b561/virtualization.h#L124
- https://github.com/Code-Hex/vz/blob/37646e120f85b7fd99989e7e4b23cf068564b561/virtualization_arm64.h#L72-L73

`convertVZMacOSRestoreImage2Struct` is the only one using `VZMacOSRestoreImage *` and not `void *`. This change makes all `convert` functions the same. And it removes a compilation warning. This will be useful for future maintainance.

Reply:
Warning is still present even if I remove `convertVZMacOSRestoreImage2Struct` from the header
```
# github.com/Code-Hex/vz/v2
virtualization_arm64.m:330:68: warning: 'VZMacOSRestoreImage' is only available on macOS 12.0 or newer [-Wunguarded-availability-new]
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Virtualization.framework/Headers/VZMacOSRestoreImage.h:30:12: note: 'VZMacOSRestoreImage' has been marked as being introduced in macOS 12.0 here, but the deployment target is macOS 11.0.0
virtualization_arm64.m:330:34: note: annotate 'convertVZMacOSRestoreImage2Struct' with an availability attribute to silence this warning
```